ravenframework/Models/PostProcessors/Validations/Representativity.py

```diff
@@ -0,0 +1,464 @@
+# limitations under the License.
+"""
+  Created on April 04, 2021
+
+  @ Authors: Mohammad Abdo  (@Jimmy-INL)
+             Congjian Wang  (@wangcj05)
+             Andrea Alfonsi (@aalfonsi)
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import numpy as np
+import xarray as xr
+import scipy as sp
+from scipy.linalg import sqrtm
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from ravenframework.utils import InputData, InputTypes
+from .. import ValidationBase
+#Internal Modules End--------------------------------------------------------------------------------
+
+class Representativity(ValidationBase):
+  """
+    Representativity is a base class for validation problems
+    It represents the base class for most validation problems
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, specs, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    specs = super(Representativity, cls).getInputSpecification()
+    prototypeParameters = InputData.parameterInputFactory("prototypeParameters", contentType=InputTypes.StringListType,
+                      descr=r"""mock model parameters/inputs""")
+    prototypeParameters.addParam("type", InputTypes.StringType)
+    specs.addSub(prototypeParameters)
+    targetParameters = InputData.parameterInputFactory("targetParameters", contentType=InputTypes.StringListType,
+                            descr=r"""Target model parameters/inputs""")
+    specs.addSub(targetParameters)
+    targetPivotParameterInput = InputData.parameterInputFactory("targetPivotParameter", contentType=InputTypes.StringType,
+                                descr=r"""ID of the temporal variable of the target model. Default is ``time''.
+        \nb Used just in case the  \xmlNode{pivotValue}-based operation  is requested (i.e., time dependent validation).""")
+    specs.addSub(targetPivotParameterInput)
+    return specs
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'POSTPROCESSOR Representativity'
+    self.dynamicType = ['static'] #  for now only static is available
+    self.name = 'Representativity'
+    self.stat = [None, None]
+    self.featureDataObject = None
+    self.targetDataObject = None
+    self.senPrefix = 'sen'
+
+  def getBasicStat(self):
+    """
+      Get Basic Statistic PostProcessor
+      @ In, None
+      @ Out, stat, object, Basic Statistic PostProcessor Object
+    """
+    from .. import factory as ppFactory # delay import to allow definition
```

Comment:
updated.

Reply:
I just reverted the change. It can not moved to the beginning of the file since the class has not been initialized yet. We need to first initialize the PP class then we can import the factory.